rust/parquet/src/data_type.rs

```diff
@@ -461,21 +553,16 @@ impl AsBytes for str {
 }
 
 pub(crate) mod private {
-    use super::{AsBytes, Result, ParquetError};
+    use crate::encodings::decoding::PlainDecoderDetails;
+    use crate::util::bit_util::{BitWriter, BitReader};
+    use crate::util::memory::ByteBufferPtr;
 
-    pub type BitIndex = u64;
+    use byteorder::ByteOrder;
+    use std::convert::TryInto;
 
-    #[derive(Copy, Clone)]
-    pub enum EncodedValue<'a> {
```

Comment:
Performance notes:

This type and its behaviour introduces branch misses that _unfortunately_ impact performance. Since removing this pushes the logic down towards the implementation type `ParquetValueType` it becomes needless (and actually I think more messy code).

As such in pushing the logic down towards the base types this goes away